File: 백준/Silver/11651. 좌표 정렬하기 2/좌표 정렬하기 2.py
import sys
input = sys.stdin.readline

N = int(input())
points = []

for _ in range(N):
    x, y = map(int, input().split())
    points.append((x, y))

points.sort(key=lambda p: (p[1], p[0]))

for p in points:
    print(p[0], p[1])

File: 백준/Silver/11651. 좌표 정렬하기 2/좌표 정렬하기 2.js
const input = require('fs').readFileSync(0).toString().trim().split('\n');

const N = input.shift();
const points = input.map(line => {
    const [x, y] = line.split(/\s+/).map(Number);
    return {x, y};
});
points.sort((a, b) => (a.y - b.y) || (a.x - b.x));

console.log(points.map(p => `${p.x} ${p.y}`).join('\n'));